src/compiler/types.ts

```diff
@@ -1780,12 +1863,30 @@ namespace ts {
         kind: SyntaxKind.ClassExpression;
     }
 
-    export interface ClassElement extends RealDeclaration {
+    export interface ClassElement extends DeclarationBase {

```

Comment:
I have wanted to make `ClassElement` and `TypeElement` discriminated unions (and replace `ObjectLiteralElement` with `ObjectLiteralElementLike`), but I've held off due to performance-related issues with respect to discriminated unions.